libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2558,21 +2573,20 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 	templ("funSel", IRVariable(_functionCall.expression()).part("functionSelector").name());
 	templ("address", IRVariable(_functionCall.expression()).part("address").name());
 
-	// Always use the actual return length, and not our calculated expected length, if returndatacopy is supported.
-	// This ensures it can catch badly formatted input from external calls.
-	if (m_context.evmVersion().supportsReturndata())
-		templ("returnSize", "returndatasize()");
-	else
-		templ("returnSize", to_string(returnInfo.estimatedReturnSize));
-
-	templ("reservedReturnSize", returnInfo.dynamicReturnSize ? "0" : to_string(returnInfo.estimatedReturnSize));
+	solAssert(
+		!(!m_context.evmVersion().supportsReturndata() && returnInfo.dynamicReturnSize),
+		"supportsReturnData and dynamicReturnSize cannot be both true"
+	);
```

Comment:
Is the following semantically equivalent?
```suggestion
	if (returnInfo.dynamicReturnSize)
		solAssert(m_context.evmVersion().supportsReturndata());
```